### DISCUSSION TITLE
ch1/ch1-01

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#127] [General]
# ch1/ch1-01

<The Go Programming Language>中文版

https://gopl-zh.github.io/ch1/ch1-01.html

**Comment by CaiZhuoyue** [comment]
现在已经不支持go get指令了耶

**Reply by wagaaa** [reply]
go install xx.xx/xx/xx

**Reply by qbmiller** [reply]
哪不支持 go get?

**Reply by Ayusummer** [reply]
> [Deprecation of 'go get' for installing executables - The Go Programming Language](https://go.dev/doc/go-get-install-deprecation)

---

```bash
'go get' is no longer supported outside a module.
        To build and install a command, use 'go install' with a version,
        like 'go install example.com/cmd@latest'
        For more information, see https://golang.org/doc/go-get-install-deprecation
        or run 'go help get' or 'go help install'.
```

go get 在 `g.mod` 中同时用于更新依赖和安装命令。这种组合很混乱，使用起来也很不方便，因为开发人员不想同时进行更新和安装。

`1.17.1` 及其后版本不再支持 `go get` 命令

如果要在当前模块的上下文中安装可执行文件时，使用 `go install` 不带版本后缀

```bash
go install example.com/cmd
```

这个命令适用于安装当前目录或父目录中go.mod定义的版本要求和其他命令。

---

要安装可执行文件同时忽略当前模块go.mod，使用go install带上版本后缀例如

```bash
go install example.com/cmd@latest
```

**Reply by jiechen257** [reply]
“以法令方式规定标准的代码格式可以避免无尽的无意义的琐碎争执（译注：也导致了 Go 语言的 TIOBE 排名较低，因为缺少撕逼的话题）”

这句话读来真喜感🤣

**Comment by newarabbit** [comment]
现在不用再下载goimports了吧，SDK自带了

**Comment by KanekiWei** [comment]
给一个建议，可不可以加一个标注功能，这样方便查看重点

**Comment by fan4w** [comment]
真的，c++的格式化不统一很烦人
